pkg/machines/components/vm/memModal.jsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import {
+    Alert,
+    Button,
+    Form,
+    FormGroup,
+    Grid,
+    Modal
+} from 'patternfly-react';
+import PropTypes from 'prop-types';
+
+import cockpit from 'cockpit';
+import * as Select from 'cockpit-components-select.jsx';
+import {
+    units,
+    convertToUnit,
+    digitFilter,
+    toFixedPrecision,
+    vmId
+} from '../../helpers.es6';
+import {
+    setMem,
+    setMaxMem
+} from '../../actions/provider-actions.es6';
+
+import './memModal.css';
+
+const _ = cockpit.gettext;
+
+const clamp = (value, max, min) => {
+    return value < min ? min : (value > max ? max : value);
+};
+
+export class MemModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showModal: false,
+            dialogError: undefined,
+            /* What user might understand as 'memory' is defined in libvirt as 'current memory' */
+            memory: convertToUnit(props.vm.currentMemory || 0, 'KiB', 'MiB'),
+            overviewMemory: convertToUnit(props.vm.currentMemory || 0, 'KiB', 'MiB'),
+            memoryUnit: units.MiB.name,
+            /* What user might understand as 'max memory' is defined in libvirt as 'memory' */
+            maxMemory: props.vm.memory || 0,
```

Comment:
@Antique please let us know what default value it makes sense to have here in case memory element is missing. 0 Doesn't sounds correct here.

Reply:
So the '0' value is a safeguard in case there is no memory or currentMemory in the domain XML? That should not never happen, libvirt will not accept XML without memory element and will add currentMemory element automatically based on memory element.  If we want to keep the safeguard we can use 1024 as default, that is used while creating new guest.